src/utilities/routes.browser.spec.ts

```diff
@@ -163,7 +163,58 @@ test('given route with equal matches, returns route with highest score', () => {
   ] as const satisfies Routes
 
   const resolved = resolveRoutes(routes)
-  const response = routeMatch(resolved, '/')
+  const response = getRouterRouteForUrl(resolved, '/')
 
   expect(response?.name).toBe('second-route')
-})
\ No newline at end of file
+})
+
+test('given a route without params or query returns an empty params and query', () => {
+  const route = {
+    name: 'route',
+    path: '/',
+    component,
+  } as const satisfies Route
+
+  const resolved = resolveRoutes([route])
+  const response = getRouterRouteForUrl(resolved, '/')
+
+  expect(response?.params).toMatchObject({})
+  expect(response?.query).toMatchObject({})
+})
+
+test('given a url with a query returns all query values', () => {
+  const route = {
+    name: 'route',
+    path: '/',
+    component,
+  } as const satisfies Route
+
+  const resolved = resolveRoutes([route])
+  const response = getRouterRouteForUrl(resolved, '/?foo=foo1&foo=foo2&bar=bar&baz')
+
+  expect(response?.query.get('foo')).toBe('foo1')
+  expect(response?.query.getAll('foo')).toMatchObject(['foo1', 'foo2'])
+  expect(response?.query.get('bar')).toBe('bar')
+  expect(response?.query.getAll('bar')).toMatchObject(['bar'])
+  expect(response?.query.get('baz')).toBe('')
+  expect(response?.query.getAll('baz')).toMatchObject([''])
+  expect(response?.query.get('does-not-exist')).toBe(null)
+  expect(response?.query.getAll('does-not-exist')).toMatchObject([])
+})
+
+test('given a route with params returns all params', () => {
+  const route = {
+    name: 'route',
+    path: '/:paramA',
+    query: 'paramB=:paramB',
+    component,
+  } as const satisfies Route
+
+  const resolved = resolveRoutes([route])
+  const response = getRouterRouteForUrl(resolved, '/A?paramB=B')
+
+  expect(response?.params).toMatchObject({
+    paramA: 'A',
+    paramB: 'B',
+  })
```

Comment:
this test is failing because of a bug in finding param values. I thinks the value for `paramA` is `A?paramB=B`